### PR TITLE
Add gear to display

### PR DESCRIPTION
**Pull request by zaphodbeblebrox**
Displays gear and amounts are editable. Also can add individual pieces of gear.

### CODE REVIEW

File: client/src/components/campaignDisplay/GearTab.tsx [hunk 1]
```diff
@@ -18,8 +20,33 @@ const GearTab = ({ campaignData, dbRefetch }: GearTabProps) => {
...
+                buttonText="Add Gear"
+                labelText="Search Gear"
+                options={Object.keys(gearData)}
+                onSubmit={(gearToAdd) => {
+                    console.log(gearToAdd);
+                    const newGearLocation = gearData[gearToAdd as keyof GearKeys].location;
+                    const updatedGear = { ...campaignData.gear };
+                    if (!campaignData.gear[newGearLocation]) {
+                        updatedGear[newGearLocation] = { [gearToAdd]: 0 };
+                    } else if (campaignData.gear[newGearLocation] && !campaignData.gear[newGearLocation][gearToAdd]) {
```
Comment (definitelynotagoblin) [inline client/src/components/campaignDisplay/GearTab.tsx]: && !campaignData.gear[newGearLocation] is redundant here

File: client/src/components/primitiveComponents/SearchBoxes.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import Fuse from 'fuse.js';
+
+interface TwilightSearchPopup {
+    buttonText: string;
+    labelText: string;
+    options: string[];
+    onSubmit: (selectedValue: string) => void;
+}
+
+const TwilightSearchPopup = ({ buttonText, labelText, options, onSubmit }: TwilightSearchPopup) => {
```
Comment (vermentor) [inline client/src/components/primitiveComponents/SearchBoxes.tsx:14]: Is there a reason to be passing options into this component?

File: client/src/components/primitiveComponents/SearchBoxes.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    // TODO: Make this popup window a static size
+    const [searchTerm, setSearchTerm] = useState('');
+    const [results, setResults] = useState<string[]>([]);
+
+    const settings = {
+        includeScore: true,
+        threshold: 0.1,
+    };
+
+    const fuse = useMemo(() => new Fuse(options, settings), []);
```
Comment (definitelynotagoblin) [inline client/src/components/primitiveComponents/SearchBoxes.tsx]: I would use `options` as a dependency on this.

File: client/src/components/campaignDisplay/GearTab.tsx (hunk 1 above)
Comment (zaphodbeblebrox) [reply client/src/components/campaignDisplay/GearTab.tsx]: deleted campaignData.gear[newGearLocation] && instead

File: client/src/components/primitiveComponents/SearchBoxes.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                <Button>{buttonText}</Button>
+            </Dialog.Trigger>
+
+            <Dialog.Content className="DialogContent">
+                <Dialog.Title className="DialogTitle">{labelText}</Dialog.Title>
+
+                <Dialog.Description className="DialogDescription">Search:</Dialog.Description>
+                <Flex direction="column" gap="3">
+                    <TextField.Input
+                        placeholder="Find gear..."
```
Comment (definitelynotagoblin) [inline client/src/components/primitiveComponents/SearchBoxes.tsx]: You should probably remove the placeholder since this is gonna be generic

**Review by definitelynotagoblin — APPROVED** [review]

**Pull request merged**